sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IdentityColumn.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.util
+
+import org.apache.spark.sql.connector.catalog.{Identifier, TableCatalog, TableCatalogCapability}
+import org.apache.spark.sql.errors.QueryCompilationErrors
+import org.apache.spark.sql.types.{StructField, StructType}
+
+case class IdentityColumnSpec(start: Long, step: Long, allowExplicitInsert: Boolean)
+
+/**
+ * This object contains utility methods and values for Identity Columns
+ */
+object IdentityColumn {
+  val IDENTITY_INFO_START = "identity.start"
```

Comment:
Does the metadata key need to be consistent with delta? https://github.com/delta-io/delta/blob/100cc4dba42ce9a88999c6c0b4fb31400bfaf0cb/spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSourceUtils.scala#L40